aws/request/request.go

```diff
@@ -565,30 +564,28 @@ func AddToUserAgent(r *Request, s string) {
 	r.HTTPRequest.Header.Set("User-Agent", s)
 }
 
+type temporary interface {
+	Temporary() bool
+}
+
 func shouldRetryCancel(r *Request) bool {
-	awsErr, ok := r.Error.(awserr.Error)
-	timeoutErr := false
-	errStr := r.Error.Error()
-	if ok {
+	temporaryError := false
+	err := r.Error
+	if awsErr, ok := err.(awserr.Error); ok {
 		if awsErr.Code() == CanceledErrorCode {
 			return false
 		}
-		err := awsErr.OrigErr()
-		netErr, netOK := err.(net.Error)
-		timeoutErr = netOK && netErr.Temporary()
-		if urlErr, ok := err.(*url.Error); !timeoutErr && ok {
```

Comment:
`net.Error` and `*url.Error` both implement temporary here; so the test can be simplified to a `Temporary()?` check

Reply:
Temporary check is good, but I think this wasn't added until Go 1.6. Since the SDK still supports back to Go 1.5 we'll need a way of detecting to  cancel for retry..

Reply:
Ah, I didn't think of this ! Thanks for pointing it.
`url.Error` has a Temporary func since go 1 but `*net.Error` was only added in go 1.6.

So, I added a case using the `Err` field in `*net.Error` that should only happen in go 1.6

Reply:
Is it okay now ?

Reply:
Thanks for the update @azr, it looks like the unit test on Go 1.5 are failing, because the SDK is unable to determine the request was canceled, and is retrying.

Reply:
Ah sorry I thought the ci was broken for some wrong reason; it was me.

Will try to check if the error returned in the test is the error returned by such a case. I have a hunch it could be another type and then the type assertion should work.
Now I need to see how to play with an old version of Go.